packages/ld-key-pair/KeyPair.ts

```diff
@@ -0,0 +1,103 @@
+/**
+ * defines the ability to sign based on a KeyPair or only
+ * from a private key
+ */
+interface Signer {
+    privateKey: string;
+    keyPair?: KeyPair;
+    sign(options: {
+        data: any;
+    }): string;
+}
+
+/**
+ * defines the ability to perform verification based on a KeyPair
+ * or a public key
+ */
+interface Verifier {
+    publicKey: string;
+    keyPair?: KeyPair;
+    verify(options: {
+        data: any;
+        signature: string;
+    }): boolean;
+}
+
+/**
+ * public key side of a key pair
+ */
+interface PublicNode {
+    id: string;
+    type: any;
+    controller?: any;
+    publicKey?: string;
+    publicKeyBase58?: string;
+}
+/**
+ * public key side of a key pair
+ */
+ interface PrivateNode {
+    id: string;
+    type: any;
+    controller?: any;
+    privateKey?: string;
+    privateKeyBase58?: string;
+}
+
+/**
+ * defines the common properties and methoods in use by 
+ * key pairs in the LD realm based on existing implementations
+ */
+interface KeyPair {
+    id: string;
+    type: any;
+    controller?: any;
+    publicKey?: string;
+    privateKey?: string;
+    publicKeyBase58?: string;
+    privateKeyBase58?: string;
+    publicNode(): PublicNode;
+    generate(options: any): KeyPair;
+    fingerprint(): string;
+    verifyFingerprint(options: {
+        fingerprint: string
+    }): boolean;
+    fromFingerprint(options: {
+        fingerprint: string
+    }): KeyPair;
+    fingerprintFromPublicKey(options: {
+        publicKeyBase58: string
```

Comment:
again a bad assumption, we expect "PublicKey" to be of type "PublicKeyNode" where "PublicKeyNode" has publicKeyJwk, or publicKeyMultibase or publicKeyBase58

Reply:
agreed - was matching the existing impls that tend towards that direction, have now made more generic with a specified encoding that can have a default (or be detected off of multibase, etc.)